tests/ci/merge_pr.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+
+"""Script to check if PR is mergeable and merge it"""
+
+import argparse
+import logging
+
+from get_robot_token import get_best_robot_token
+from github_helper import GitHub, PullRequest
+from pr_info import PRInfo
+
+
+def parse_args() -> argparse.Namespace:
+    pr_info = PRInfo()
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description="Script to download binary artifacts from S3. Downloaded artifacts "
+        "are renamed to clickhouse-{static_binary_name}",
+    )
+    parser.add_argument(
+        "--check-approved",
+        action="store_true",
+        help="if set, checks that the PR is approved and no changes required",
+    )
+    parser.add_argument(
+        "--repo",
+        default=pr_info.repo_full_name,
+        help="PR number to check",
+    )
+    parser.add_argument(
+        "--pr",
+        type=int,
+        default=pr_info.number,
+        help="PR number to check",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def is_approved(pr: PullRequest) -> bool:
+    logging.info("Checking the PR for approvals")
+    reviews = pr.get_reviews()
+    if not reviews.totalCount:
+        logging.info("There aren't reviews for PR #%s", pr.number)
+        return False
+    # We get reviews in a chronological order, so we'll have the only latest
+    # review per user
+    review_per_user = {r.user: r.state for r in reviews}
+    if "CHANGES_REQUESTED" in review_per_user.values():
+        logging.info(
+            "The following users requested changes for the PR: %s",
+            ", ".join(
+                user.login
+                for user, state in review_per_user.items()
+                if state == "CHANGES_REQUESTED"
+            ),
+        )
+        return False
+
+    if "APPROVED" in review_per_user.values():
```

Comment:
Right now anybody can approve PR, for example me, but I don't have write access.
So `review_per_user` should be checked for write access, right?
Or maybe it works like this already and I'm missing something?

Reply:
Also it seems that it does not check that there were no changes after review? (like issue describes)
But maybe project had been reconfigured and now approve will be removed after new changes.

Actually most of the time subsequent changes may indeed fix failed checks correctly, and I'm pretty sure that there will be the case when I would wish for such behavior, but tricky cases are possible...

Reply:
can you really approve a PR? it sounds suspicious

The second comment makes perfect sense, and `is_approved` should check for changes after approval

Reply:
>can you really approve a PR?

As you can see below

>[azat](https://github.com/azat) approved these changes [2 days ago](https://github.com/ClickHouse/ClickHouse/pull/41110#pullrequestreview-1103124112)

And it is not only me, but anyone, since default policy is to allow anyone - https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-pull-request-reviews-in-your-repository

But indeed it has different color though.